py3status/modules/mpd_status.py

```diff
@@ -9,41 +9,67 @@
 @license Eclipse Public License
 """
 
-import re
+import string
 from mpd import (MPDClient, CommandError)
 from socket import error as SocketError
 from time import time
 
-# state characters (or strings). Actual of them replaces {state} placeholder in self.format
-STATE_CHARACTERS = {
-    "pause": "[pause]",
-    "play": "[play]",
-    "stop": "[stop]",
-}
+
+class _DefaultFormatter(string.Formatter):
+    """
+    Custom implementation of string Formatter that returns
+    a default string of raising KeyError on missing
+    fields.
+    """
+    def __init__(self, default_value=""):
+        self._default_value = default_value
+
+    def get_value(self, *args, **kwargs):
+        try:
+            return super().get_value(*args, **kwargs)
+        except KeyError:
+            return self._default_value
 
 
 class Py3status:
     """
     Configuration parameters:
         - format : indicator text format
+        - format_fallback : fallback indicator text format (see below)
+        - state_play : label to display for "playing" state
+        - state_pause : label to display for "paused" state
+        - state_stop : label to display for "stopped" state
+        - fallback_if_empty : comma-separated list of metadata (see below)
         - hide_when_paused / hide_when_stopped : hide any indicator, if
         - host : mpd host
         - max_width : if text length will be greater - it'll shrink it
         - password : mpd password
         - port : mpd port
 
-    Format of result string can contain:
-        {state} - current state from STATE_CHARACTERS
+    `format` parameter can contain:
+        {state} - current state (see state_{play,pause,stop} parameters)
         Track information:
-        {track}, {artist}, {title}, {time}, {album}, {pos}
-        In additional, information about next track also comes in,
-        in analogue with current, but with next_ prefix, like {next_title}
+        {track}, {artist}, {title}, {time}, {album}, {pos}, {file}
+        In addition, information about the next track are also available,
+        similar to the above but prefixed with next_ as in: {next_title}
+    
+    `fallback_if_empty` contains a comma-separated list of the above blocks.
+        Example: 'artist,title'
+        If all these metadata are empty, `format_fallback` is used instead
+        of the usual `format`. This is useful if your songs are missing
+        metadata so you can display something else than an empty string.
+        Set this parameter to an empty string to disable the feature.
     """
 
     # available configuration parameters
     cache_timeout = 2
     color = None
     format = '{state} №{pos}. {artist} - {title} [{time}] | {next_title}'
+    format_fallback = '{state} №{pos}. {file} [{time}] | {next_title}'
+    fallback_if_empty = 'artist,title'
+    state_play = "[play]"
+    state_pause = "[pause]"
+    state_stop = "[stop]"
```

Comment:
Just to comment here, customisable state strings would be very useful, since you can put play (">"), pause, stop icons in their place, rather than the textual representations of the states.